SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -290,6 +290,12 @@ public void drainPendingQueue(final SingularitySchedulerStateCache stateCache) {
           scheduledTasks += scheduleTasks(stateCache, maybeRequest.get().getRequest(), requestState,
               deployStatistics, pendingRequest, matchingTaskIds, maybePendingDeploy);
           requestManager.deletePendingRequest(pendingRequest);
+        } else if (updatedRequest.isScheduled()
+            && (pendingRequest.getPendingType() == PendingType.NEW_DEPLOY
+            || pendingRequest.getPendingType() == PendingType.TASK_DONE)) {
+          // If we are here, there is already an immediate of run of the scheduled task launched. Drop anything that would
```

Comment:
Is is possible to write a unit test to prove out this particular case? i.e. what happens if the immediate is first vs the NEW_DEPLOY/TASK_DONE comes first in the queue